FileConv/Gfx/PrintShopFont.cs

```diff
@@ -0,0 +1,416 @@
+                ResourceMgr? resMgr, ConvFlags convFlags, AppHook appHook)
+                : base(attrs, dataStream, rsrcStream, resMgr, convFlags, appHook)
+        {
+            Applic = TestApplicability();
+        }
+
+        protected override Applicability TestApplicability()
+        {
+            if (DataStream == null)
+            {
+                return Applicability.Not;
+            }
+
+            if (FileAttrs.FileType == FileAttribs.FILE_TYPE_BIN)
+            {
+                if (FileAttrs.DataLength <= MAX_FONT_FILE_SIZE &&
+                   ((FileAttrs.AuxType == 0x6000 && FileAttrs.DataLength >= MINIMUM_HEADER_SIZE) ||
+                    (FileAttrs.AuxType == 0x5FF4 && FileAttrs.DataLength >= MINIMUM_HEADER_SIZE + PSC_FONT_HEADERS_SIZE)))
+                {
+                    UnpackData();
+
+                    if (DataPointersAreValid())
+                    {
+                        if (CountOversizedCharacters() == 0)
+                        {
+                            return Applicability.Yes;
+                        }
+                        else
+                        {
+                            return Applicability.Probably;
+                        }
+                    }
+                }
+            }
+            return Applicability.Not;
+        }
+
+        public override Type GetExpectedType(Dictionary<string, string> options)
+        {
+            return typeof(IBitmap);
+        }
+
+        public override IConvOutput ConvertFile(Dictionary<string, string> options)
+        {
+            if (Applic <= Applicability.Not)
+            {
+                Debug.Assert(false);
+                return new ErrorText("ERROR");
+            }
+
+            Debug.Assert(DataBuf != null);
+            Debug.Assert(DataBuf.Length >= 0);
+            Debug.Assert(DataStream != null);
+
+
+            bool doMult = GetBoolOption(options, OPT_MULT, false);
+
+            return ConvertMono(DataBuf, Palette8.Palette_MonoBW, doMult);
+        }
+
+        /// <summary>
+        /// Converts a Print Shop font file to a B&amp;W bitmap.
+        /// </summary>
+        public Bitmap8 ConvertMono(byte[] buf, Palette8 palette, bool doMult)
+        {
+            Bitmap8 output;
+
+            int totalWidth = 0;
+            int maxHeight = 0;
+
+            for (int idx = 1; idx < CHARACTER_COUNT; idx++)
+            {
+                if (idx != 32)
+                {
+                    if (idx != 1) totalWidth += INTER_CHARACTER_SPACING_PX;
+                    totalWidth += GetWidthForCharacter(idx);
+                    int height = GetHeightForCharacter(idx);
+                    if (height > maxHeight) { maxHeight = height; }
+                }
+            }
+
+            Debug.WriteLine("TotalWidth: " + totalWidth);
+
+            if (doMult)
+            {
+                output = new Bitmap8((4 + totalWidth) * XMULT, (4 + maxHeight) * YMULT);
```

Comment:
Is the "4 +" for padding?

Reply:
Yes.  I replaced the literal 4s with constants for LEFT_PADDING, RIGHT_PADDING, TOP_PADDING, and BOTTOM_PADDING throughout.